src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,372 @@
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object CakeTracker {
+
+    private fun getCakeTrackerData() = ProfileStorageData.profileSpecific?.cakeTracker
+    private val config get() = SkyHanniMod.feature.inventory.cakeTracker
+    private var currentYear = 0
+
+    private var inCakeBag = false
+    private var inCakeInventory = false
+    private var timeOpenedCakeInventory: Long = 0
+    private var inAuctionHouse = false
+    private var unobtainedCakesDisplayed = false
+    private var searchingForCakes = false
+    private var knownCakesInCurrentInventory = mutableListOf<Int>()
+
+    /**
+     * REGEX-TEST: §cNew Year Cake (Year 360)
+     * REGEX-TEST: §cNew Year Cake (Year 1,000)
+     * REGEX-TEST: §f§f§cNew Year Cake (Year 330)
+     */
+    private val cakeNamePattern by patternGroup.pattern(
+        "cake.name",
+        "(?:§f§f)?§cNew Year Cake \\(Year (?<year>[\\d,]*)\\)",
+    )
+
+    /**
+     * REGEX-TEST: Ender Chest (2/9)
+     * REGEX-TEST: Jumbo Backpack (Slot #6)
+     * REGEX-TEST: New Year Cake Bag
+     */
+    private val cakeContainerPattern by patternGroup.pattern(
+        "cake.container",
+        "^(Ender Chest \\(\\d{1,2}/\\d{1,2}\\)|.*Backpack(?:§r)? \\(Slot #\\d{1,2}\\)|New Year Cake Bag)$",
+    )
+
+    /**
+     * REGEX-TEST: New Year Cake Bag
+     */
+    private val cakeBagPattern by patternGroup.pattern(
+        "cake.bag",
+        "^New Year Cake Bag$",
+    )
+
+    /**
+     * REGEX-TEST: Auctions Browser
+     * REGEX-TEST: Auctions: "Test"
+     */
+    private val auctionBrowserPattern by patternGroup.pattern(
+        "auction.search",
+        "^(Auctions Browser|Auctions: \".*)$",
+    )
+
+    /**
+     * REGEX-TEST: Auctions: "New Year C
+     */
+    private val auctionCakeSearchPattern by patternGroup.pattern(
+        "auction.cakesearch",
+        "^Auctions: \"New Year C.*$",
+    )
+
+    private val tracker = SkyHanniTracker("New Year Cake Tracker", { Data() }, { it.cakeTracker })
+    { drawDisplay(it) }
+
+    class Data : TrackerData() {
+        override fun reset() {
+            cakesOwned.clear()
+        }
+
+        @Expose
+        var cakesOwned: MutableList<Int> = mutableListOf()
+
+        @Expose
+        var cakesMissing: MutableList<Int> = mutableListOf()
+    }
+
+    private fun addCake(cakeYear: Int) {
+        val cakeTrackerData = getCakeTrackerData() ?: return
+        if (!cakeTrackerData.cakesOwned.contains(cakeYear)) {
+            tracker.modify {
+                it.cakesOwned.add(cakeYear)
+            }
+        }
+        recalculateMissingCakes()
+    }
+
+    private fun removeCake(cakeYear: Int) {
+        val cakeTrackerData = getCakeTrackerData() ?: return
+        if (cakeTrackerData.cakesOwned.contains(cakeYear)) {
+            tracker.modify {
+                it.cakesOwned.remove(cakeYear)
+            }
+        }
+        recalculateMissingCakes()
+    }
+
+    private fun isEnabled(): Boolean {
```

Comment:
please just do
```suggestion
    private fun isEnabled(): Boolean = LorenzUtils.inSkyBlock && config.enabled
```
(idk how to format suggestions so that it also removes the other lines of the function)